utils.js

```diff
@@ -213,10 +233,17 @@ module.exports = {
    * @param {object} props  - The request-basis properties, which affect cache behavior.
    * @returns {object} The modified properties.
    */
-  handleReqCacheHeaders: function(req, props) {
+  handleReqCacheHeaders: function (req, props) {
     const cacheControl = req.get('cache-control');
-    if (typeof cacheControl === 'string' && cacheControl.toLowerCase().indexOf('only-if-cached') !== -1) {
-      props.doQuery = false;
+    if (typeof cacheControl === 'string') {
+      if (cacheControl.toLowerCase().indexOf('only-if-cached') !== -1) {
+        props.doQuery = false;
+      }
+      // the expiration can also be set via the Request header.
+      const maxAgeMatch = cacheControl.toLowerCase().match(/^(.*max-age=)(\d*).*$/);
```

Comment:
If the header is `max-age=`, `maxAgeMatch` will be `''` and `props.expiration` `NaN`. That would cause problems, right?